sky/provision/kubernetes/volume.py

```
@@ -231,21 +232,84 @@ def map_all_volumes_usedby(
+    if isinstance(capacity, dict) and 'storage' in capacity:
+        size_quantity = capacity['storage']
+    # Fallback to spec.resources.requests (dict) - requested size
+    if size_quantity is None:
+        requests = getattr(getattr(pvc_obj.spec, 'resources', None), 'requests',
+                           None)
+        if isinstance(requests, dict):
+            size_quantity = requests.get('storage')
+    # Parse and normalize the size if found
+    if size_quantity:
+        try:
+            # Normalize to GB string (e.g., '20')
+            pvc_size = resources_utils.parse_memory_resource(
+                size_quantity, 'size', allow_rounding=True)
+        except ValueError as e:
+            # Just log the error since it is not critical.
+            logger.warning(f'Failed to parse PVC size {size_quantity!r} '
+                           f'for PVC {pvc_name}: {e}')
+    if pvc_size is not None:
+        if config.size is not None and config.size != pvc_size:
+            logger.warning(f'PVC {pvc_name} has size {pvc_size} but config '
+                           f'size is {config.size}, overriding the config size'
+                           f' with the PVC size.')
+        config.size = pvc_size
+
+
+def create_persistent_volume_claim(
+    namespace: str,
+    context: Optional[str],
+    pvc_spec: Dict[str, Any],
+    config: Optional[models.VolumeConfig] = None,
+) -> None:
     """Creates a persistent volume claim for SkyServe controller."""
     pvc_name = pvc_spec['metadata']['name']
     try:
-        kubernetes.core_api(context).read_namespaced_persistent_volume_claim(
-            name=pvc_name, namespace=namespace)
+        pvc = kubernetes.core_api(
+            context).read_namespaced_persistent_volume_claim(
+                name=pvc_name, namespace=namespace)
+        if config is not None:
+            _populate_config_from_pvc(config, pvc)
         logger.debug(f'PVC {pvc_name} already exists')
```

Shall we raise an error if `use-existing=false` in this case?